libtenzir/include/tenzir/split_regex.hpp

```diff
@@ -0,0 +1,50 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/chunk.hpp"
+#include "tenzir/generator.hpp"
+
+#include <boost/regex.hpp>
+
+#include <optional>
+#include <string_view>
+
+namespace tenzir {
+
+inline auto split_regex(std::string separator) {
+  return [separator = std::move(separator)](generator<chunk_ptr> input)
+           -> generator<std::optional<std::string_view>> {
+    auto sep = separator;
+    auto buffer = std::string{};
+    auto expr = boost::regex{sep, boost::regex_constants::no_except};
```

Comment:
In all other places we use PCRE, either directly or through Arrow. Mind changing to that?

Reply:
That's not PCRE but re2. That library does not support positive lookaround, which is an essential feature to make this optional useful in practice. The example given in the PR description would not work with re2.